Katello/katello
# Fixes #32435 - errata srpms not considering during filtering

Fixes an issue where SRPMs were not being considering when deciding which errata to copy to a new repository.  If the erratum had SRPMs in it, the erratum would never be copied because the repository would appear to be missing the erratum's packages.

To test:

1) Sync the filesystem "zoo" repository that I've updated to have an SRPM-only erratum
  -> Go to the repo on your filesystem, run `./create` (remove the rpm2swidtag call from the script if you must), and host it with `python -m SimpleHTTPServer <port>`.  Then you can sync.
2) Create a content view and add the zoo repo to it
3) Create any kind of filter as long as it affects the zoo repository
4) Publish and see that the "RHEA-2021:9999" erratum did not make it into the content view version
5) Patch in this PR and try again

Consider trying the above steps with and without depsolving.  Also, consider trying it with the original problem repositories, the EPEL ones (https://dl.fedoraproject.org/pub/epel/7/).  ~The only annoying thing about EPEL is that you need to sync a number of repositories for errata filtering to work, since errata have packages spread around all the repos.~ (Fixed in https://github.com/Katello/katello/pull/9615)
